Automattic/wp-calypso
# Implement useSubscriptionManagerUserSettingsMutation

Closes https://github.com/Automattic/wp-calypso/issues/74433

## Proposed Changes

This PR makes saving the `UserSettings` form possible. It adds:

- The mutation 
- The button
- A notice component for success and error states

## Testing Instructions

1. Apply this PR
2. Visit http://calypso.localhost:3000/subscriptions/settings
3. ⚠️⚠️⚠️ Clear your indexedDb from the Devtools & reload! Otherwise cached query data may still be present.  ⚠️⚠️⚠️

![CleanShot 2023-03-23 at 10 16 15@2x](https://user-images.githubusercontent.com/528287/227157336-ea99a39f-86f0-4b98-b003-13aa0937d2af.png)

3. Make some changes
4. Click the blue save button
5. You should get a notice that the settings are saved.

<img width="1601" alt="CleanShot 2023-03-23 at 10 21 20@2x" src="https://user-images.githubusercontent.com/528287/227158367-82c1f67f-e90d-4112-b996-69ecbba3160f.png">

** Testing the error state **
Add an invalid input field in `packages/subscription-manager/src/components/fields/EmailFormatInput/EmailFormatInput.tsx`

```jsx
<option value="invalid">{ translate( 'Invalid' ) }</option>
```

Try saving this value:

<img width="1513" alt="CleanShot 2023-03-23 at 10 23 11@2x" src="https://user-images.githubusercontent.com/528287/227158887-3dcd1c95-f84e-4912-9cdc-d420a081006f.png">


## Pre-merge Checklist
